packages/browsers/src/fileUtil.ts

```diff
@@ -105,19 +92,60 @@ function createXzStream() {
       return stream.destroy(e);
     });
 
+  child.once('close', () => {
+    return stream.end();
+  });
+
   return stream;
 }
 
 /**
  * @internal
  */
-function extractTarXz(tarPath: string, folderPath: string): Promise<void> {
-  return new Promise((fulfill, reject) => {
-    const tarStream = tar.extract(folderPath);
-    tarStream.on('error', reject);
-    tarStream.on('finish', fulfill);
-    const readStream = createReadStream(tarPath);
-    readStream.pipe(createXzStream()).pipe(tarStream);
+export const internalConstantsForTesting = {
+  xz: 'xz',
+  bzip2: 'bzip2',
+};
+
+/**
+ * @internal
+ */
+async function extractTar(
+  tarPath: string,
+  folderPath: string,
+  decompressUtilityName: keyof typeof internalConstantsForTesting,
+): Promise<void> {
+  const tarFs = await import('tar-fs');
+  return await new Promise<void>((fulfill, reject) => {
+    function handleError(utilityName: string) {
+      return (error: Error) => {
+        if ('code' in error && error.code === 'ENOENT') {
+          error = new Error(
+            `\`${utilityName}\` utility is required to unpack this archive`,
+            {
+              cause: error,
+            },
+          );
+        }
+        reject(error);
+      };
+    }
+    const unpack = spawn(
+      internalConstantsForTesting[decompressUtilityName],
+      ['-d'],
+      {
+        stdio: ['pipe', 'pipe', 'inherit'],
+      },
+    )
+      .once('error', handleError(decompressUtilityName))
+      .once('exit', code => {
+        debugFileUtil(`${decompressUtilityName} exited, code=${code}`);
```

Comment:
Why don't we use handleError here?

Reply:
There is no error provided to the exit event.